assets/js/custom/ProgramCredits.js

```diff
@@ -1,51 +1,217 @@
 
-    const url = Routing.generate('edit_program_credits', { id: programId, new_credits: newCredits }, false)
+    const languageSelected = creditsSelect.value
+    if (languageSelected === '' || languageSelected === 'default') {
+      const url = Routing.generate('edit_program_credits', { id: programId, new_credits: newCredits }, false)
 
-    $.get(url, function (data) {
-      if (parseInt(data.statusCode) === 200) {
-        location.reload()
-      } else if (parseInt(data.statusCode) === 707) {
-        editCredits.addClass('danger')
-        editCreditsError.show()
-        editCreditsError.text(data.message)
+      $.get(url, function (data) {
+        if (parseInt(data.statusCode) === 200) {
+          location.reload()
+        } else if (parseInt(data.statusCode) === 707) {
+          editCredits.addClass('danger')
+          editCreditsError.show()
+          editCreditsError.text(data.message)
+        }
+      })
+    } else if (newCredits === '') {
+      customTranslationApi.deleteCustomTranslation(
+        programId,
+        languageSelected,
+        deleteCustomTranslationSuccess,
+        showCreditsError
+      )
+    } else {
+      customTranslationApi.saveCustomTranslation(
+        programId,
+        newCredits,
+        languageSelected,
+        saveCustomTranslationSuccess,
+        showCreditsError
+      )
+    }
+  }
+
+  function keepOrDiscardChangesResult (result) {
+    if (result.isConfirmed) {
+      previousCreditsIndex = creditsSelect.selectedIndex
+      lastSavedCredits = credits.text().trim()
+    } else if (result.isDenied) {
+      previousCreditsIndex = creditsSelect.selectedIndex
+      getCustomTranslation()
+    } else if (result.isDismissed) {
+      creditsSelect.selectedIndex = previousCreditsIndex
+    }
+  }
+
+  function closeEditorDialogResult (result) {
+    if (result.isConfirmed) {
+      saveCredits()
+    } else if (result.isDenied) {
+      closeCreditsEditor()
+    }
+  }
+
+  function areChangesSaved () {
+    return editCredits.val().trim() === lastSavedCredits.trim()
+  }
+
+  function resetCreditsEditor () {
+    creditsSelect.selectedIndex = 0
+    previousCreditsIndex = 0
+    const defaultCredits = credits.text().trim()
+    editCredits.val(defaultCredits)
+    lastSavedCredits = defaultCredits
+  }
+
+  function closeCreditsEditor () {
+    descriptionCreditsContainer.show()
+    descriptionHeadline.show()
+    editCreditsUI.addClass('d-none')
+    resetCreditsEditor()
+    handleShowMore()
+  }
+
+  editCreditsButton.on('click', () => {
+    descriptionCreditsContainer.hide()
+    descriptionHeadline.hide()
+    editCreditsUI.removeClass('d-none')
+    showMoreToggle.addClass('d-none')
+  })
+
+  closeCreditsEditorButton.on('click', () => {
+    if (areChangesSaved()) {
+      closeCreditsEditor()
+    } else {
+      closeEditorDialog.show(closeEditorDialogResult)
+    }
+  })
+
+  editCreditsSubmitButton.on('click', () => {
+    saveCredits()
+  })
+
+  if (myProgram) {
```

Comment:
do we really need this? isn't the edit button not visible if not logged in or not my program? The api does check for logging in before making changes

Reply:
Since the language selector is not included when it is not the user's program the creditSelect will be undefined and then this `creditSelect.listen` function throws an error. So, if the function isn't called when it isn't the user's program then the error won't be thrown. Same thing for the description

Reply:
Since the language selector is not included when it is not the user's program the creditSelect will be undefined and then this `creditSelect.listen` function throws an error. So, if the function isn't called when it isn't the user's program then the error won't be thrown. Same thing for the description

Reply:
ok